simnibs/simulation/fem.py

```diff
@@ -1463,10 +1462,22 @@ def tms_many_simulations(
             E = np.vstack([-d.dot(v) for d in D]).T * 1e3
             dAdt = dAdt[roi]
             E -= dAdt
-            if field == 'E':
-                out_field = E
-            elif field == 'J':
-                out_field = calc_J(E, cond)
+
+            # build output fields
+            out_field = []
+            if 'E' in field:
+                out_field.append(E)
+            if 'D' in field:
+                out_field.append(dAdt)
+            if 'J' in field:
+                out_field.append(calc_J(E, cond))
+            if 'v' in field:
+                out_field.append(v)
+            out_field = tuple(out_field)
+
+            # if only one field to output, un-tuple
+            if len(out_field) == 1:
+                out_field = out_field[0]
             if post_pro is not None:
                 out_field = post_pro(out_field)
             with h5py.File(fn_hdf5) as f:
```

Comment:
I guess calling tms_many_simulations() with the new feature of multiple fields (e.g. with field='ED') but without a post_pro function would crash here. To circumvent, one would probably have to change the structure of fn_hdf5 here and everywhere.